File: ada/ada2012/Cpp/Symbol.cpp
#include "Symbol.h"
#include <sstream>

std::string Symbol::toString() const {
    std::string result = name + " (with classification ";
    bool first = true;
    for (auto tc : classification) {
        if (!first) result += ", ";
        result += std::to_string(static_cast<int>(tc));
        first = false;
    }
    result += ")";
    if (isComposite) result += " [composite]";
    if (!definedFile.empty()) {
        result += " defined at " + definedFile + ":" + std::to_string(definedLine) + ":" + std::to_string(definedColumn);
    }
    return result;
}

File: ada/ada2022/Go/Symbol.go
package parser

import (
	"fmt"
	"strings"
)

type Symbol struct {
	Name           string
	Classification map[TypeClassification]bool
	Members        map[string]*Symbol
	Parent         *Symbol
	Predefined     bool
	IsComposite    bool
	DefinedFile    string
	DefinedLine    int
	DefinedColumn  int
}

func NewSymbol() *Symbol {
	return &Symbol{
		Classification: make(map[TypeClassification]bool),
		Members:        make(map[string]*Symbol),
	}
}

func (s *Symbol) String() string {
	var classifications []string
	for tc := range s.Classification {
		classifications = append(classifications, fmt.Sprintf("%d", tc))
	}
	result := s.Name + " (with classification " + strings.Join(classifications, ", ") + ")"
	if s.IsComposite {
		result += " [composite]"
	}
	if s.DefinedFile != "" {
		result += fmt.Sprintf(" defined at %s:%d:%d", s.DefinedFile, s.DefinedLine, s.DefinedColumn)
	}
	return result
}
